src/main/java/org/opensearch/knn/index/query/BaseQueryFactory.java

```diff
@@ -90,20 +100,71 @@ protected static Query getFilterQuery(BaseQueryFactory.CreateQueryRequest create
                 createQueryRequest.getFieldName()
             )
         );
+
+        // preserve nestedStack
+        Deque<ObjectMapper> nestedLevelStack = new LinkedList<>();
+        ObjectMapper objectMapper = null;
+        while ((objectMapper = queryShardContext.nestedScope().getObjectMapper()) != null) {
+            nestedLevelStack.push(objectMapper);
+            queryShardContext.nestedScope().previousLevel();
```

Comment:
Can we call this in line 107 to set the object mapper?

Reply:
No, because `NestedScope#levelStack` is a `Deque` object. when call `previousLevel` directly in while condition, it would throw `NoSuchElementException` when deque is empty.